edx/edx-enterprise
# [ENT-744] Track enterprise course enrollment events.

**Description:** This patch adds `track_event` calls to all call sites where `EnterpriseCourseEnrollment` instances get created.

**JIRA:** [ENT-744](https://openedx.atlassian.net/browse/ENT-744)

**Dependencies:** None

**Testing instructions:**

1. Install this patch & run the migration.
2. Create some enterprise course enrollments (direct from admin, pending from admin, course landing page...).
4. Verify tracking events are emitted whenever course enrollment objects get created. One way to do that is to temporarily patch the `enterprise.utils.track_event` method to log its calls.

**Merge checklist:**

  **DO NOT** just add dependencies to `requirements/*.txt` files.

**Post merge:**

**Author concerns:** 

There are many different flows and it is hard to capture all of them without threading a lot of parameters through multiple systems edx-enterprise, edx-platform, ecommerce. For example an enrollment may start on the course landing page, go via ecommerce and finally be created on the data sharing consent page.
